cpp/benchmarks/ndsh/q1.cpp

```diff
@@ -100,15 +102,13 @@
   return charge;
 }
 
-int main(int argc, char const** argv)
+void run_ndsh_q1(nvbench::state& state)
 {
-  auto const args = parse_args(argc, argv);
-
-  // Use a memory pool
-  auto resource = create_memory_resource(args.memory_resource_type);
-  rmm::mr::set_current_device_resource(resource.get());
+  double const scale_factor = state.get_float64("scale_factor");
 
-  cudf::examples::timer timer;
+  // Define a map for holding parquet data sources
+  std::unordered_map<std::string, parquet_device_buffer> sources;
+  generate_parquet_data_sources(scale_factor, {"lineitem"}, sources);
```

Comment:
is this data generation required for q1? 
Data generation should not be part of measured time. It should be part of prior code. Please check other benchmarks as reference.

Reply:
Wow, this was a huge blunder from my side. Fixing it. Thanks a lot for catching it @karthikeyann